### DISCUSSION TITLE
fixed time period on/off

### DISCUSSION THREAD

**Discussion by JaDiko** [arendst/Tasmota#22547] [Support request]
Hello,
I would like to ask for help in solving a problem.
I want to have a TASMOTA 8266
(https://bilder.obi.sk/93aaded6-d082-4945-8b53-8957f5f059e8/prZZO/5543228.jpg)
set so that when I plug it in, it turns on after 15 minutes, stays on for 3 minutes and then turns off. This is how it should work 24/7.
I don't want any solution using MQTT or anything else.
I just want a "magic" rule or "something" that will allow me to do this.
The above socket wakes up the modem, wifi for 3 minutes after 15 minutes and sends me the measured values, but then I need the modem, wifi, probes to be off.
Don't tell me that such a sophisticated device can't handle it, I guess it's not just worse than this:
https://media.hornbach.sk/hb/packshot/as.46488068.jpg?dvid=8
because that's also a solution
Thanks for the help.

**Comment by lalo-uy** [comment]
Just set pulsetime to 300 sec ( 5min) and a rule triggered on boot to 900
sec that set power on

PulseTime 300
Rule1 ON system#boot do RuleTimer1 900 ENDON
ON Rules#Timer=1 DO Power1 on ENDON

**Reply by sfromis** [reply]
If you want it to repeat, change the last rule to
`ON Rules#Timer=1 DO Backlog Power1 on; RuleTimer1 900 ENDON`
Otherwise it is once-only.

Beware that without Wifi, Tasmota is not fully initialized, and timing may not work well, but using `System#Init` instead of `System#Boot` lets it start even before full initialization.

**Comment by JaDiko** [comment]
Hi, 
I tried to enter it through the console, but it doesn't work.
Are you sure this is correct?
I'm running 12.3.1(tasmota), but there is no problem upgrading.
On the console it looks like this:
**PulseTime** Ent
_RESULT = {"PulseTime":{"Set":[60,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"Remaining":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]}}_
**Rule1** Ent
RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":38,"Free":473,"Rules":"ON system#boot do RuleTimer1 120 ENDON"}}
**ON Rules#Timer=1 DO Backlog Power1 on; RuleTimer1 900 ENDON** Ent
_RESULT = {"Command":"Unknown"}_
I've shortened the time, but the third set doesn't work.

I'm not very good at this, so I'd like to ask for help.
Thanks

**Reply by sfromis** [reply]
`Rule1` is the command for rule set 1, and `on ... endon` is the content; has to be one line, including multiple rules, each of which are `on ... endon`. Both rules are on the same line as the `Rule1` command.

Also, after inputting the rule set, you'll have to activate it by another command, `Rule1 1`

**Comment by JaDiko** [comment]
Hello,
 I'm struggling with your advice, but I can't.
I'm probably making a mistake,
I'm attaching my statement:

22:20:15.135 CMD: PulseTime 300
22:20:15.139 RSL: RESULT = {"PulseTime1":{"Set":300,"Remaining":299}}
22:20:21.125 CMD: PulseTime
22:20:21.134 RSL: RESULT = {"PulseTime":{"Set":[300,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"Remaining":[294,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]}}
22:20:33.500 CMD: ON Rules#Timer=1 DO Backlog Power1 on; RuleTimer1 900 ENDON
22:20:33.506 RSL: RESULT = {"Command":"Unknown","Input":"ON Rules#Timer=1 DO Backlog Power1 on; RuleTimer1 900 ENDON"}
22:20:37.553 CMD: rule
22:20:37.559 RSL: RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
22:20:42.684 CMD: rule 1
22:20:42.690 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
22:20:45.535 CMD: rule
22:20:45.541 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

What state should the relay be in after inserting it, ON or OFF.
The function should be such that I insert the plug and after 15 minutes TASMOTA turns on the relay, which will hold for 5 minutes and then turns off and after 15 minutes turns on again for 5 minutes - continuous alternation.
I'm testing on:
https://tasmota-github-io.translate.goog/docs/devices/OBI-Socket-2/?_x_tr_sl=auto&_x_tr_tl=sk&_x_tr_hl=sk
14.3.0 (release-tasmota) 
Thanks in advance

**Reply by sfromis** [reply]
Rule sets are entered using the `Rule` command
https://tasmota.github.io/docs/Commands/#rule
And `on` is the start of each rule, within a rule set.

**Reply by barbudor** [reply]
And entering the ruleset in 1 single command line:
```
Rule1 on....... endon
```
From the 'on' of the 1st rule to the 'endon' of the last rule in the set
